[NETBEANS-4619][CND] Missing license headers

vieiro opened the pull request:
Adding  missing license headers in different CND modules. In my box this clears the RAT report for the CND cluster.
Some cnd modules do not support FreeMarker templates, this can be improved in the future once CND compiles.

matthiasblaesing commented on cnd/cnd.asm/licenseinfo.xml:
```diff
@@ -0,0 +1,31 @@
+        <file>src/org/netbeans/modules/cnd/asm/core/resources/AsmTemplate.s</file>
+        <file>src/org/netbeans/modules/cnd/asm/core/resources/InlineTemplate.il</file>
+        <license ref="Apache-2.0-ASF" />
+        <comment type="TEMPLATE_MINIMAL_IP" />
```
I think `COMMENT_UNSUPPORTED` would be more appropriate here.

matthiasblaesing commented on cnd/cnd.editor/licenseinfo.xml:
```diff
@@ -0,0 +1,32 @@
+        <file>src/org/netbeans/modules/cnd/editor/resources/cplusplus/HExample</file>
+        <file>src/org/netbeans/modules/cnd/editor/resources/fortran/FExample</file>
+        <license ref="Apache-2.0-ASF" />
+        <comment type="TEMPLATE_MINIMAL_IP" />
```
I think `GUI_USABILITY` would be more appropriate

matthiasblaesing commented on cnd/cnd.remote/tools/todo.txt:
```diff
@@ -1,3 +1,20 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
I would use the small ALv2 header here (https://www.apache.org/legal/src-headers.html#is-a-short-form-of-the-source-header-available)

matthiasblaesing commented on cnd/cnd.ui/src/org/netbeans/modules/cnd/resources/templates/qt/dialog-buttonsbottom.template:
```diff
@@ -1,3 +1,23 @@
+<!--
```
Are these user templates? If so, this looks as if the license header is taken over into the generated file. I would not add a header then.

matthiasblaesing commented on cnd/cnd.ui/src/org/netbeans/modules/cnd/resources/templates/qt/dialog-buttonsright.template:
```diff
@@ -1,3 +1,23 @@
+<!--
```
Are these user templates? If so, this looks as if the license header is taken over into the generated file. I would not add a header then.

matthiasblaesing commented on cnd/cnd.ui/src/org/netbeans/modules/cnd/resources/templates/qt/dialog-nobuttons.template:
```diff
@@ -1,3 +1,23 @@
+<!--
```
Are these user templates? If so, this looks as if the license header is taken over into the generated file. I would not add a header then.

matthiasblaesing commented on cnd/cnd.ui/src/org/netbeans/modules/cnd/resources/templates/qt/mainwindow.template:
```diff
@@ -1,3 +1,23 @@
+<!--
```
Are these user templates? If so, this looks as if the license header is taken over into the generated file. I would not add a header then.

matthiasblaesing commented on cnd/cnd.ui/src/org/netbeans/modules/cnd/resources/templates/qt/resource.template:
```diff
@@ -1,4 +1,24 @@
 <?xml version="1.0" encoding="utf-8"?>
+<!--
```
Are these user templates? If so, this looks as if the license header is taken over into the generated file. I would not add a header then.

matthiasblaesing commented on cnd/cnd.ui/src/org/netbeans/modules/cnd/resources/templates/qt/translation.template:
```diff
@@ -1,4 +1,24 @@
 <?xml version="1.0" encoding="utf-8"?>
+<!--
```
Are these user templates? If so, this looks as if the license header is taken over into the generated file. I would not add a header then.

matthiasblaesing commented on cnd/cnd.ui/src/org/netbeans/modules/cnd/resources/templates/qt/widget.template:
```diff
@@ -1,3 +1,23 @@
+<!--
```
Are these user templates? If so, this looks as if the license header is taken over into the generated file. I would not add a header then.

vieiro commented:
Hi @matthiasblaesing , thanks for your review! I've removed the license from the QT files and added a licenseinfo.xml for these files with GUI_USABILITY. Let me know if this is correct.

matthiasblaesing reviewed (approved):
Looks sane to me.

vieiro commented:
Thanks, @matthiasblaesing. Let me learn all about the surprsing "Squash and merge" magic button before hitting it...

matthiasblaesing commented:
@vieiro my personal experience: Don't trust the github facitilities for anything more complex than a plain merge. Github has the tendency to destroy history:

- If you don't expose your full name in your github settings, the author information will be broken by squash done by github
- If you don't expose at least one e-mail address in your github settings as public visible, the author information will equally be broken

vieiro commented:
@matthiasblaesing yes, I've read your understandable concerns about this in the mailing list.
I talked with ASF Infra about this and they suggested talking to dev@community.apache.org instead (https://mail-archives.apache.org/mod_mbox/community-dev//202104.mbox/browser), I've just emailed them.  Let's see what they have to say.

vieiro commented:
D'oh! It seems the public name/email address in github has to be set _before_ the PR is done. Sorry by that!

neilcsmith-net commented:
We might consider the .mailmap file option as well? Should tidy up the history.

vieiro commented:
@matthiasblaesing The fact is that I did have a public name/email address available, but it seems this has to be public _before_ the PR is made somehow. Otherwise it won't be taken into account. So my level of trust with github's squash and merge is zero now.
@neilcsmith-net I'm not sure a .mailmap will help, but let me know how I can set it up and we can give it a run.

neilcsmith-net commented:
@vieiro maybe https://git-scm.com/docs/gitmailmap or check across ASF repos. Few projects have this set to map github private emails, etc. to the right person / email.
